front/src/modules/trainschedule/components/Timetable/TimetableTrainCard.tsx

```diff
@@ -50,34 +50,124 @@ function TimetableTrainCard({
+    id: train.rolling_stock_id,
+  });
   const { t } = useTranslation(['operationalStudies/scenario']);
   const dispatch = useDispatch();
 
+  const [postTrainSchedule] =
+    osrdEditoastApi.endpoints.postTrainScheduleStandaloneSimulation.useMutation();
+  const [getTrainScheduleById] = osrdEditoastApi.endpoints.getTrainScheduleById.useLazyQuery();
+  const [deleteTrainScheduleById] = osrdEditoastApi.endpoints.deleteTrainScheduleById.useMutation();
+
+  const changeSelectedTrainId = (trainId: number) => {
+    dispatch(updateSelectedTrainId(trainId));
+  };
+
   const editTrainSchedule = () => {
     dispatch(updateTrainScheduleIDsToModify([train.id]));
     setDisplayTrainScheduleManagement(MANAGE_TRAIN_SCHEDULE_TYPES.edit);
   };
 
-  useEffect(() => {
-    if (train.id) {
-      getTrainSchedule({ id: train.id })
+  const deleteTrain = async () => {
+    deleteTrainScheduleById({ id: train.id })
+      .unwrap()
+      .then(() => {
+        dispatch(
+          setSuccess({
+            title: t('timetable.trainDeleted', { name: train.train_name }),
+            text: '',
+          })
+        );
+      })
+      .catch((e: unknown) => {
+        console.error(e);
+        if (e instanceof Error) {
+          dispatch(
+            setFailure({
+              name: e.name,
+              message: e.message,
+            })
+          );
+        }
+      });
+  };
+
+  const duplicateTrain = async () => {
+    // Static for now, will be dynamic when UI will be ready
+    const trainName = `${train.train_name} (${t('timetable.copy')})`;
+    const trainDelta = 5;
+    const trainCount = 1;
+    const actualTrainCount = 1;
+
+    const trainDetail = await getTrainScheduleById({ id: train.id })
+      .unwrap()
+      .catch((e: unknown) => {
+        if (e instanceof Error) {
+          dispatch(
+            setFailure({
+              name: e.name,
+              message: e.message,
+            })
+          );
+        }
+      });
+
+    if (trainDetail) {
+      const newTrain = {
+        ...trainDetail,
+        rolling_stock: trainDetail.rolling_stock_id,
+        timetable: trainDetail.timetable_id,
+        departure_time: train.departure_time + 60 * trainDelta,
+        train_name: trainNameWithNum(trainName, actualTrainCount, trainCount),
+      };
+      await postTrainSchedule({
+        body: {
+          timetable: trainDetail.timetable_id,
+          path: trainDetail.path_id,
+          schedules: [newTrain],
+        },
+      })
         .unwrap()
-        .then((trainSchedule) => {
-          if (trainSchedule.rolling_stock_id)
-            getRollingStock({ id: trainSchedule.rolling_stock_id });
+        .then(() => {
+          refetchTimetable();
+          dispatch(
+            setSuccess({
+              title: t('timetable.trainAdded'),
+              text: `${trainName}`,
+            })
+          );
+        })
+        .catch((e: unknown) => {
+          console.error(e);
+          const error =
```

Comment:
why do we have all this new code ?

Reply:
I moved this code from the parent component (Timetable) to here because the parent component does not need it